package.json

```diff
@@ -13,7 +13,7 @@
     "lint": "eslint . --report-unused-disable-directives",
     "flow": "flow --include-warnings",
     "test": "node node_modules/jest/bin/jest.js --config=.jest.json --silent",
-    "test-ci": "yarn flow && yarn lint && yarn test",
+    "test-ci": "yarn flow && yarn lint && yarn test --runInBand",
```

Comment:
It was not necessary locally but somehow CI failed and timed out while running the tests (for no apparent reason).
https://jestjs.io/docs/en/troubleshooting.html#tests-are-extremely-slow-on-docker-and-or-continuous-integration-ci-server

Reply:
Right, here is how we went in the mobile repository to tackle this problem:

- `--runInBand` - removed because it's anti-CI
- `--maxWorkers=4` - removed because we needed even more performance
- our GitLab runners - win

Seems like this is the greatest motivation to stop creating new repositories because you have to go through all the difficulties all over again even though we did it one year ago. It's very time consuming to setup all the things properly. It's from my point of view very inefficient. 🙂

Reply:
I absolutely agree with you, but maybe we should document these choices so that if/when somebody needs to set something up, then they have a list of best practices/common issues and their possible solutions.
Could you make precise what you mean by 

> `--maxWorkers=4`- removed

as I can see it [here](https://github.com/kiwicom/mobile/blob/master/scripts/test.sh#L9)?
Are you suggesting that I remove `--runInBand`, and instead use the same flags as in the `mobile` repository, namely `--ci --color --logHeapUsage --maxWorkers=4`?

Reply:
> but maybe we should document these choices

Usually, it's not that easy to document everything and to follow it later. It's constantly evolving and more importantly: when you don't start all over again then you don't have only simple setup like this but you have really advanced tooling around (see GraphQL repo for example). It's very complicated to replicate the same thing in every repo.

I just see quite clearly the significant difference between the situation when people are starting all over again (even with the docs) and when people are investing into one thing and reusing it properly. :)

> as I can see it here?

It's a mistake.

> Are you suggesting that I remove --runInBand, and instead use the same flags as in the mobile repository, namely --ci --color --logHeapUsage --maxWorkers=4?

`--logHeapUsage` is not necessary.

Reply:
That makes sense.

> It's a mistake.

I removed it here https://github.com/kiwicom/mobile/pull/1388

I tried to follow your suggestions on this repository here https://github.com/kiwicom/margarita/pull/86

Reply:
Because CI must give you feedback fast otherwise people are being annoyed. Option `--runInBand` is quite the opposite. So one way is to improve the CI, spinup machine with more CPUs and so on and the opposite way is to get rid of parallelism and suffer. This is why I consider this option to be anti _continuous_ integration. :)